### ISSUE TITLE
Recover and finish parked serialization / type handling overhaul

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#3490]
See #3489 for background information. @eviltak is behind the parked code related to this and may be interested in continuing with it, but is low on time as of this writing. If anybody else is hugely interested in serialization and type handling feel free to look at this but be warned that it is a large piece of work and a complex topic :-)

Goal for this issue is to work with branch https://github.com/MovingBlocks/Terasology/tree/newSerialization to finish stabilizing it so it can be re-merged

As part of that work several modules need syntax changes to compile again, as was the case when originally merging #3449. The same modules and likely more beyond that will need changes after #3456 is considered as well, both for compile fixes (maybe just the same set of modules as the first PR) and runtime issues encountered in testing with the second PR (unsure if all those would go away purely with additional engine changes)

Affected modules by round one and their initial syntax fix commits (reverted as part of preparing this issue):

* https://github.com/Terasology/MasterOfOreon/commit/17a008952995a354e93f3cc2c71ed14ae8b7adec
* https://github.com/Terasology/DynamicCities/commit/bce035463cf156fc7208257f7e4adcb8bbabbf5e
* https://github.com/Terasology/Dialogs/commit/3b5804ebf557060c24a53a79760a3d3e80cf1064
* https://github.com/Terasology/Tasks/commit/b1576fb1011f8fd7886ff7a96007ce85d2895e52
* https://github.com/Terasology/TutorialDynamicCities/commit/e5766b8374f53e0d0b9a600f53aa7176f94876d1
* https://github.com/Terasology/LightAndShadow/commit/fa33f332569990f1c6cab306fe52635b402b06a5

To be clear those commits just let those modules compile after the *first* round of serialization changes. It may be pointless to re-apply those changes exactly, as the underlying changes from round two may require entirely different fixes.

See the PRs linked in #3489 for more research and discovered issues.

As a potential follow-up to this the new Record & Replay system is awaiting some of these changes to better improve its own usage of serialized events and such. @iaronaraujo would be the primary contributor involved in that effort.

**Comment by Cervator** [comment]
See also the assorted linked issues in module repos with confirmed (or possible) incompatibilities pending fixes before we can merge the new serialization.
